File: web_scheduler/app/schemas/artifact.py
from datetime import datetime

from pydantic import BaseModel


class TaskArtifactListItem(BaseModel):
    id: int
    task_id: int
    run_id: int
    artifact_type: str
    is_latest: bool
    version_no: int
    created_at: datetime

    model_config = {"from_attributes": True}


class TaskArtifactResponse(TaskArtifactListItem):
    content_text: str | None
    content_html: str | None
    content_json: str | None

File: web_scheduler/frontend/src/types/artifact.ts
export interface ArtifactListItem {
  id: number;
  task_id: number;
  run_id: number;
  artifact_type: "html" | "json" | "text" | "csv";
  is_latest: boolean;
  version_no: number;
  created_at: string;
}

export interface Artifact extends ArtifactListItem {
  content_text?: string | null;
  content_html?: string | null;
  content_json?: string | null;
}
